Firmware/cardreader.cpp

```diff
@@ -317,19 +334,17 @@ void CardReader::getAbsFilename(char *t)
  * @param[in,out] fileName
  *  expects file name including path
  *  in case of absolute path, file name without path is returned
- * @param[in,out] dir SdFile object to operate with,
- *  in case of absolute path, curDir is modified to point to dir,
- *  so it is not possible to create on stack inside this function,
- *  as curDir would point to destroyed object.
  */
-void CardReader::diveSubfolder (const char *fileName, SdFile& dir)
+bool CardReader::diveSubfolder (const char *&fileName)
```

Comment:
combining a pointer and a reference is very hard to read/verify in the code, please use a pointer to pointer if you really need an in-out parameter

Reply:
This was the initial implementation of diveSubfolder according to the comments above, so I'd prefer if I kept it like this. It's 100% C++ complinat, but not C compliant. It's good that we don't care about that scenario in this case.